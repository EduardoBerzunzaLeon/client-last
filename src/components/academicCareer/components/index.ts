export * from './StudentCard';
export * from './UserCreatorCard';
