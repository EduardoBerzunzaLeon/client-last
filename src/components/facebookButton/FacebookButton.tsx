import { useCallback } from 'react';

import { ReactFacebookLoginInfo, ReactFacebookFailureResponse } from 'react-facebook-login';
import { Toast } from 'primereact/toast';
import FacebookLogin from 'react-facebook-login/dist/facebook-login-render-props';

import { SlipButton } from '../slipButton/SlipButton';
import { useSignInWithSocialMutation } from '../../redux/auth/auth.api';
import useToast from '../../hooks/useToast';

export const FacebookButton = () => {
  const [ signInSocial, { isLoading }] = useSignInWithSocialMutation();
  const { toast, showError } = useToast();

  const responseFacebook = useCallback((
    response: ReactFacebookLoginInfo | ReactFacebookFailureResponse,
  ) => {
    if ('accessToken' in response) {
      signInSocial({
        socialName: 'facebook',
        tokenId: response.accessToken,
      });
    } else {
      showError({
        detail: 'error',
        summary: 'error',
      });
    }
  }, []);

  return (
    <>
      <Toast ref={toast} />
      <FacebookLogin
        appId={process.env.REACT_APP_CLIENT_ID_FACEBOOK || ''}
        callback={responseFacebook}
        render={(renderProps) => (
          <SlipButton
            color="indigo"
            icon="facebook"
            label="Facebook"
            isLoading={isLoading}
            onClick={renderProps.onClick}
          />
        )}
      />
    </>
  );
};

export default FacebookButton;
