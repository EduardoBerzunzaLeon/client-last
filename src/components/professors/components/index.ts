export * from './ButtonProfessorActive';
export * from './columns';
export * from './course';
export * from './ExcelButtonProfessors';
export * from './Header';
export * from './ProfessorDataForm';
export * from './ProfessorDialog';
