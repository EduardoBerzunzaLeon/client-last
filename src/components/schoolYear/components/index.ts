export * from './PhaseCard';
export * from './FileErrorsTable';
export * from './CurrentSubjectsForm';
export * from './FailedSubjectsForm';
export * from './ValidationPasswordForm';
export * from './SchoolYearStepForm';
