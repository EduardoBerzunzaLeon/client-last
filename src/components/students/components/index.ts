export * from './columns';
export * from './Header';
export * from './StudentDataForm';
export * from './StudentDialog';
export { ProfessorsHistoryDialog } from './professorsHistory';
