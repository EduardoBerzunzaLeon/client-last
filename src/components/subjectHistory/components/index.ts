export * from './SubjectsStudied';
export * from './CurrentSubjects';
export * from './UnstudySubjects';
export * from './columns';
export * from './PhaseDialog';
export * from './Header';
export * from './PhaseDataForm';
