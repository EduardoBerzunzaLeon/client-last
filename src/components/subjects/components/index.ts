export * from './Header';
export * from './columns';
export * from './SubjectDialog';
export * from './SubjectsList';
export * from './SubjectDetailDialog';
