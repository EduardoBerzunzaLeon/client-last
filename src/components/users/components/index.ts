export * from './AdminPasswordForm';
export * from './ButtonUserBlocked';
export * from './columns';
export * from './Header';
export * from './UserDataForm';
export * from './UserDialog';
