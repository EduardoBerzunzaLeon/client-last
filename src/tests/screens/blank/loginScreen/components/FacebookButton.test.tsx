import { mount } from 'enzyme';

import { FacebookButton } from '../../../../../screens/blank/loginScreen/components/FacebookButton';
import { render, Wrapper } from '../../../../fixtures/render';

const mockSignInWithSocialMutation = jest.fn();

jest.mock('../../../../../redux/auth/auth.api', () => ({
  ...jest.requireActual('../../../../../redux/auth/auth.api'),
  useSignInWithSocialMutation: () => [
    mockSignInWithSocialMutation(),
    {
      isLoading: false,
    },
  ],
}));

describe('<FacebookButton />', () => {
  let wrapper: Wrapper;
  beforeEach(() => {
    wrapper = mount(render(FacebookButton));
  });

  test('should math to snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should call signInSocial mutation', () => {
    wrapper.find('.button-slip').last().simulate('click');
    expect(mockSignInWithSocialMutation).toHaveBeenCalledTimes(1);
  });
});
