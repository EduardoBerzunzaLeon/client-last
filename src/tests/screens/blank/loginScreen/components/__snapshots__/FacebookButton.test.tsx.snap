// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FacebookButton /> should math to snapshot 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <FacebookButton>
    <Toast
      appendTo="self"
      baseZIndex={0}
      className={null}
      id={null}
      onClick={null}
      onHide={null}
      onRemove={null}
      onShow={null}
      position="top-right"
      style={null}
      transitionOptions={null}
    >
      <Portal
        appendTo="self"
        element={
          <div
            className="p-toast p-component p-toast-top-right"
            id={null}
            style={null}
          >
            <TransitionGroup
              childFactory={[Function]}
              component="div"
            />
          </div>
        }
        onMounted={null}
        onUnmounted={null}
        visible={false}
      >
        <div
          className="p-toast p-component p-toast-top-right"
          id={null}
          style={null}
        >
          <TransitionGroup
            childFactory={[Function]}
            component="div"
          >
            <div />
          </TransitionGroup>
        </div>
      </Portal>
    </Toast>
    <t
      appId="326264092782043"
      authType=""
      callback={[Function]}
      cookie={false}
      disableMobileRedirect={false}
      fields="name"
      isMobile={false}
      language="en_US"
      onFailure={null}
      redirectUri="http://localhost/"
      render={[Function]}
      responseType="code"
      returnScopes={false}
      scope="public_profile,email"
      state="facebookdirect"
      version="2.3"
      xfbml={false}
    >
      <SlipButton
        color="indigo"
        icon="facebook"
        label="Facebook"
        loading={false}
        onClick={[Function]}
      >
        <div
          className="container-slip "
        >
          <ForwardRef
            className="button-slip indigo p-0 mt-1 w-full"
            loading={false}
            onClick={[Function]}
          >
            <ButtonComponent
              badge={null}
              badgeClassName={null}
              className="button-slip indigo p-0 mt-1 w-full"
              disabled={false}
              forwardRef={null}
              icon={null}
              iconPos="left"
              label={null}
              loading={false}
              loadingIcon="pi pi-spinner pi-spin"
              onClick={[Function]}
              tooltip={null}
              tooltipOptions={null}
            >
              <button
                className="p-button p-component button-slip indigo p-0 mt-1 w-full"
                disabled={false}
                onClick={[Function]}
              >
                <i
                  className="pi pi-facebook px-2"
                />
                <span
                  className="px-3"
                >
                  Facebook
                </span>
                <Ripple />
              </button>
            </ButtonComponent>
          </ForwardRef>
        </div>
      </SlipButton>
    </t>
  </FacebookButton>
</Provider>
`;
